src/psyclone/domain/lfric/arg_ordering.py

```diff
@@ -159,6 +318,22 @@ def arglist(self):
                 f"Has the generate() method been called?")
         return self._arglist
 
+    @property
+    def psyir_arglist(self):
+        '''
+        :return: the kernel argument list as PSyIR expressions. The generate \
+            method must be called first.
+        :rtype: List[:py:class:`psyclone.psyir.nodes.Reference`]
+
+        :raises InternalError: if the generate() method has not been called.
+
+        '''
+        if not self._generate_called:
+            raise InternalError(
+                f"The PSyIR argument list in {type(self).__name__} is empty. "
```

Comment:
Well, the reason would be if we have kernels that take no argument. With my original code this would work, with checking if psyir_arglist is empty we can't support this anymore.
But obviously a kernel without arguments will not happen, so I've changed it as requested (though then to be consistent, shouldn't we remove the `_generate_called` at all, and check for empty lists everywhere??)